src/cli/Cli.ts

```diff
@@ -158,110 +158,109 @@ export class Cli {
       .then(_ => process.exit(0), err => this.closeWithError(err, optionsWithoutShorts));
   }
 
-  public static executeCommand(command: Command, args: { options: minimist.ParsedArgs }): Promise<void> {
-    return new Promise<void>((resolve: () => void, reject: (error: any) => void): void => {
-      const logger: Logger = {
-        log: (message: any): void => {
-          const output: any = Cli.formatOutput(message, args.options);
-          Cli.log(output);
-        },
-        logRaw: (message: any): void => Cli.log(message),
-        logToStderr: (message: any): void => Cli.error(message)
-      };
-
-      if (args.options.debug) {
-        logger.logToStderr(`Executing command ${command.name} with options ${JSON.stringify(args)}`);
-      }
-
-      // store the current command name, if any and set the name to the name of
-      // the command to execute
-      const cli = Cli.getInstance();
-      const parentCommandName: string | undefined = cli.currentCommandName;
-      cli.currentCommandName = command.getCommandName(cli.currentCommandName);
+  public static async executeCommand(command: Command, args: { options: minimist.ParsedArgs }): Promise<void> {
+    const logger: Logger = {
+      log: (message: any): void => {
+        const output: any = Cli.formatOutput(message, args.options);
+        Cli.log(output);
+      },
+      logRaw: (message: any): void => Cli.log(message),
+      logToStderr: (message: any): void => Cli.error(message)
+    };
 
-      command.action(logger, args as any, (err: any): void => {
-        // restore the original command name
-        cli.currentCommandName = parentCommandName;
+    if (args.options.debug) {
+      logger.logToStderr(`Executing command ${command.name} with options ${JSON.stringify(args)}`);
+    }
 
-        if (err) {
-          return reject(err);
-        }
+    // store the current command name, if any and set the name to the name of
+    // the command to execute
+    const cli = Cli.getInstance();
+    const parentCommandName: string | undefined = cli.currentCommandName;
+    cli.currentCommandName = command.getCommandName(cli.currentCommandName);
 
-        if (args.options.debug || args.options.verbose) {
-          const chalk: typeof Chalk = require('chalk');
-          logger.logToStderr(chalk.green('DONE'));
-        }
+    try {
+      await command.action(logger, args as any);
+    }
+    finally {
+      // restore the original command name
+      cli.currentCommandName = parentCommandName;
+    }
 
-        resolve();
-      });
-    });
+    if (args.options.debug || args.options.verbose) {
```

Comment:
If I'm not mistaken, here we're changing behavior: where in the past 'DONE' would be displayed only when command succeeded, now it's displayed all the time. If that's indeed the case, let's keep the original behavior to avoid introducing breaking changes.

Reply:
If you look at the original code:

```js
      command.action(logger, args as any, (err: any): void => {
        // restore the original command name
        cli.currentCommandName = parentCommandName;

        if (err) {
          return reject(err);
        }

        if (args.options.debug || args.options.verbose) {
          const chalk: typeof Chalk = require('chalk');
          logger.logToStderr(chalk.green('DONE'));
        }

        resolve();
      });
```

we use `return` with `reject` so in case of an error, the DONE statement isn't printed.

To keep this behavior, we could use:

```js
    try {
      await command.action(logger, args as any);

      if (args.options.debug || args.options.verbose) {
        const chalk: typeof Chalk = require('chalk');
        logger.logToStderr(chalk.green('DONE'));
      }
    }
    finally {
      // restore the original command name
      cli.currentCommandName = parentCommandName;
    }
```

This way, if there's an exception in executing the command, we'll skip the 'DONE' statement and continue to the finally clause to restore the name.

Reply:
I think this comes down to the same thing, but your version looks better/more logical. I will adjust it!